pkg/adapter/adapter.go

```diff
@@ -175,35 +171,40 @@ func (a *Adapter) start(stopCh <-chan struct{}) error {
 		zap.String("Name", a.config.Name),
 		zap.String("Namespace", a.config.Namespace))
 
-	conn, err := a.CreateConn(a.config.User, a.config.Password, logger)
-	if err == nil {
-		defer conn.Close()
+	if a.config.User != "" && a.config.Password != "" {
+		a.config.Brokers = fmt.Sprintf(
+			"amqp://%s:%s@%s",
+			a.config.User,
+			a.config.Password,
+			vhostHandler(a.config.Brokers, a.config.Vhost),
+		)
 	}
 
-	ch, err := a.CreateChannel(conn, nil, logger)
+	a.setupRabbitMQ()
 	if err == nil {
-		defer ch.Close()
+		defer a.connection.Close()
```

Comment:
I'd sugges to have a new function like `shutdownRabbitMQ` and handle resource cleanup here, otherwise these defers are a bit confuse since nor connection nor channel are not setup here. So they 'leak' here in a way.